kin-devplatform-sdk/src/main/java/kin/devplatform/data/blockchain/BlockchainSourceImpl.java

```diff
@@ -424,6 +421,12 @@ public void updateActiveAccount(int accountIndex) throws BlockchainException {
 		getBalance(null);
 	}
 
+	@Override
+	public void updateKinClientAndAccount(IKinClient kinClient, int accountIndex) {
+		this.kinClient = kinClient;
```

Comment:
see other related comments, it's not enough, you need to update index and to update balance like in `updateActiveAccount`, I suggest to merge the 2 methods

Reply:
Did it already.
Thanks.